tests/connect_redirect/connect_redirect_tests.cpp

```diff
@@ -0,0 +1,433 @@
+        REQUIRE(family == AF_INET);
+        v4_addresses++;
+    }
+    if (_vip_v4 != "") {
+        get_address_from_string(_vip_v4, _globals.addresses[socket_family_t::IPv4].vip_address, false, &family);
+        REQUIRE(family == AF_INET);
+        get_address_from_string(_vip_v4, _globals.addresses[socket_family_t::Dual].vip_address, true, &family);
+        REQUIRE(family == AF_INET);
+        v4_addresses++;
+    }
+    REQUIRE((v4_addresses == 0 || v4_addresses == 3));
+
+    IN4ADDR_SETLOOPBACK((PSOCKADDR_IN)&_globals.addresses[socket_family_t::IPv4].loopback_address);
+    IN6ADDR_SETV4MAPPED(
+        (PSOCKADDR_IN6)&_globals.addresses[socket_family_t::Dual].loopback_address,
+        &in4addr_loopback,
+        scopeid_unspecified,
+        0);
+
+    // Read v6 addresses.
+    if (_remote_ip_v6 != "") {
+        get_address_from_string(
+            _remote_ip_v6, _globals.addresses[socket_family_t::IPv6].remote_address, false, &family);
+        REQUIRE(family == AF_INET6);
+        v6_addresses++;
+    }
+    if (_local_ip_v6 != "") {
+        get_address_from_string(_local_ip_v6, _globals.addresses[socket_family_t::IPv6].local_address, false, &family);
+        REQUIRE(family == AF_INET6);
+        v6_addresses++;
+    }
+    if (_vip_v6 != "") {
+        get_address_from_string(_vip_v6, _globals.addresses[socket_family_t::IPv6].vip_address, false, &family);
+        REQUIRE(family == AF_INET6);
+        v6_addresses++;
+    }
+    REQUIRE((v6_addresses == 0 || v6_addresses == 3));
+    IN6ADDR_SETLOOPBACK((PSOCKADDR_IN6)&_globals.addresses[socket_family_t::IPv6].loopback_address);
+
+    _globals.destination_port = _destination_port;
+    _globals.proxy_port = _proxy_port;
+    _globals_initialized = true;
+}
+
+static void
+_load_and_attach_ebpf_programs(_Outptr_ struct bpf_object** return_object)
+{
+    struct bpf_object* object = bpf_object__open("cgroup_sock_addr2.o");
+    REQUIRE(object != nullptr);
+    REQUIRE(bpf_object__load(object) == 0);
+
+    bpf_program* connect_program_v4 = bpf_object__find_program_by_name(object, "connect_redirect4");
+    REQUIRE(connect_program_v4 != nullptr);
+
+    int result = bpf_prog_attach(
+        bpf_program__fd(const_cast<const bpf_program*>(connect_program_v4)), 0, BPF_CGROUP_INET4_CONNECT, 0);
+    REQUIRE(result == 0);
+
+    bpf_program* connect_program_v6 = bpf_object__find_program_by_name(object, "connect_redirect6");
+    REQUIRE(connect_program_v6 != nullptr);
+
+    result = bpf_prog_attach(
+        bpf_program__fd(const_cast<const bpf_program*>(connect_program_v6)), 0, BPF_CGROUP_INET6_CONNECT, 0);
+    REQUIRE(result == 0);
+
+    *return_object = object;
+}
+
+static void
+_update_policy_map(
+    _In_ const struct bpf_object* object,
+    _In_ sockaddr_storage& destination,
+    _In_ sockaddr_storage& proxy,
+    uint16_t destination_port,
+    uint16_t proxy_port,
```

Comment:
why can't you just use the port field inside the destination and proxy sockaddrs?

Reply:
Same as above. We can set the port in destination and proxy sockaddrs and then pass to this (and other) function, but it will need to be done before every new test.